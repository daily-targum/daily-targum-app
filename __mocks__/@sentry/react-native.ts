export const init = () => jest.fn();