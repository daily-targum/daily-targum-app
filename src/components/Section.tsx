import React from 'react';
import { useStyleCreator, makeStyleCreator } from './Theme';
import { View, ViewStyle } from 'react-native';

function Section({
  children,
  style,
  innerStyle,
  testID
}: {
  children: React.ReactNode,
  style?: ViewStyle | ViewStyle[],
  innerStyle?: ViewStyle | ViewStyle[],
  testID?: string
}) {
  const styles = useStyleCreator(styleCreator);
  return (
    <View 
      testID={testID} 
      style={[styles.sectionWrap, style]}
    >
      <View style={[styles.section, innerStyle]}>
        {children}
      </View>
    </View>
  )
}

const styleCreator = makeStyleCreator(theme => ({
  sectionWrap: {
    alignItems: 'center'
  },

  section: {
    flex: 1,
    padding: theme.spacing(2),
    maxWidth: 750,
    width: '100%'
  }
}));

export default Section;