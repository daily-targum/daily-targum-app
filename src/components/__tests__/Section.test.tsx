jest.mock('../Theme');

import React from 'react';
import renderer from 'react-test-renderer';
import Section from '../Section';
import { Text } from 'react-native';

describe('<Section />', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <Section>
        <Text>Hello World!!!</Text>
      </Section>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
