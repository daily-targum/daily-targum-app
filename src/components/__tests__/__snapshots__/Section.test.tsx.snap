// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "maxWidth": 750,
          "padding": 14,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text>
      Hello World!!!
    </Text>
  </View>
</View>
`;
